File: frontend/src/Models/Blog.ts
import { User } from "./User";

export type Blog = {
    id:string;
    type:BlogType;
    title:string;
    content:string;
    tags:string[];
    heroImage:string;
    images:string[];
    author:User;
    createdAt:Date;
    updatedAt:Date;
}

enum BlogType {
    PRODUCT = "PRODUCT",
    BLOG = "BLOG",
}

File: backend/src/main/java/com/shraj/backend/models/Blog.java
package com.shraj.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UuidGenerator;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Blog  extends Auditable {

    @Id
    @GeneratedValue
    @UuidGenerator
    @Column(name = "id", nullable = false, updatable = false, length = 36)
    private String id;

    @Column(name = "type")
    private String type;

    private String title;

    private String content;

    private String[] tags;

    private String heroImage;

    private String[] images;

    private Boolean isDeleted;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User author;

}
